code/clue.js

```diff
@@ -1,59 +1,317 @@
 const pickKiller = () => {
-  // This will randomly select a killer from the suspects. And add that to the mystery object.
-  mystery.killer = randomSelector(suspects)
-
-  // This will change the background color of the card to the one connected to the chosen killer and show the full name of the killer. Feel free to add more things to show about the killer.
-  document.getElementById('killerCard').style.background = mystery.killer.color
-  document.getElementById(
-    'killerName'
-  ).innerHTML = `${mystery.killer.firstName} ${mystery.killer.lastName}`
+  mystery.killer = randomSelector(suspects)   // This will randomly select a killer from the suspects. And add that to the mystery object.
+  document.getElementById('killerCard').style.background = mystery.killer.color   // This will change the background color of the card to the one connected to the chosen killer and show the full name of the killer.
+  document.getElementById('killerName').innerHTML = `Name: ${mystery.killer.firstName} ${mystery.killer.lastName}`
+  document.getElementById('killerAge').innerHTML = `Age: ${mystery.killer.age}`
+  document.getElementById('killerOccupation').innerHTML = `Occupation: ${mystery.killer.occupation}`
+  document.getElementById('killerImage'); killerImage.src = mystery.killer.image
+  document.getElementById('killerDescription').innerHTML = `${mystery.killer.description}`;
 }
+document.getElementById('killerCard').onclick = pickKiller;
 
-// CREATE FUNCTIONS pickWeapon and pickRoom in a similar way.
 
-// STEP 4 - CREATE A FUNCTION revealMystery that will be invoked when you click that button. It should show something like:
-// 'The murder was committed by Jacob Green, in the living room with a rope.'
+// This function will be invoked when you click on the weapon card.
+const pickWeapon = () => {
+  mystery.weapon = randomSelector(weapons)
+  document.getElementById('weaponCard').style.background = mystery.weapon.color
+  document.getElementById('weaponName').innerHTML = `${mystery.weapon.name}`
+  document.getElementById('weaponWeight').innerHTML = `Weight: ${mystery.weapon.weight}`
+  document.getElementById('weaponLenght').innerHTML = `Lenght: ${mystery.weapon.length}`
+  document.getElementById('weaponImage'); weaponImage.src = mystery.weapon.image;
+}
+document.getElementById('weaponCard').onclick = pickWeapon;
+
+
+// This function will be invoked when you click on the room card.
+const pickRoom = () => {
+  mystery.room = randomSelector(rooms)   
+  document.getElementById('roomCard').style.background = mystery.room.color
+  document.getElementById('roomName').innerHTML = `${mystery.room.name}`
+  document.getElementById('roomImage'); roomImage.src = mystery.room.image;
+}
+document.getElementById('roomCard').onclick = pickRoom;
+
+// FUNCTION FOR REVEALING MYSTERY //
+//Function to reveal the mystery and verify that all cards are picked
+const revealMystery  = () => {
+  if(mystery.killer && mystery.weapon && mystery.room) {
```

Comment:
Very good statement :)
for further information about how does it work, please read https://www.freecodecamp.org/news/js-type-coercion-explained-27ba3d9a2839/

Moreover, you need to fix indentation in this statement, right now it's wrong